server/src/main/java/org/elasticsearch/search/aggregations/metrics/GeoCentroidAggregationBuilder.java

```diff
@@ -62,8 +67,7 @@ protected GeoCentroidAggregationBuilder(GeoCentroidAggregationBuilder clone, Bui
 
     @Override
     protected ValuesSourceType defaultValueSourceType() {
-        // TODO: This should probably not be BYTES, but we're not failing tests with BYTES, so needs more tests?
-        return CoreValuesSourceType.BYTES;
+        return CoreValuesSourceType.GEOPOINT;
```

Comment:
I still don't understand why this was passing with BYTES, but it's probably okay.   GeoPoint is the correct default.